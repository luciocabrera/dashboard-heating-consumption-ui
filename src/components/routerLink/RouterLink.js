// React
import React from 'react';
// Router
import { Link } from 'react-router-dom';

const RouterLink = ({ href, ...props }) => <Link to={href} {...props} />;

export default RouterLink;
