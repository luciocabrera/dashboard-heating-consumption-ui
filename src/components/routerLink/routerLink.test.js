// /* globals describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { Router } from 'react-router-dom';
import RouterLink from './RouterLink';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory({ basename: process.env.PUBLIC_URL });

describe('Rendering RouterLink ', () => {
  it('RouterLink  renders using ReactDOM without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router history={history}>
        <RouterLink href='/' />
      </Router>,
      div
    );
  });
  it(
    'RouterLink  renders using enzyme shallow without crashing',
    () => {
      shallow(
        <Router history={history}>
          <RouterLink href='/' />
        </Router>
      );
    }
  );
  it(
    'RouterLink renders using enzyme mount without crashing',
    () => {
      const wrapper = mount(
        <Router history={history}>
          <RouterLink href='/' />
        </Router>
      );
      expect(wrapper.exists()).toBe(true);
    }
  );
});

describe('RouterLink (Snapshot)', () => {
  it('RouterLink  renders using renderer without crashing', () => {
    const component = renderer.create(
      <Router history={history}>
        <RouterLink href='/' />
      </Router>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
