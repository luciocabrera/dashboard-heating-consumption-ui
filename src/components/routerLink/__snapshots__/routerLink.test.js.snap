// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterLink (Snapshot) RouterLink  renders using renderer without crashing 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;
